sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,163 @@
+func (gem *globalEndpointManager) GetWriteEndpoints() ([]url.URL, error) {
+	return gem.locationCache.writeEndpoints()
+}
+
+// GetReadEndpoints returns the read endpoints from the location cache.
+func (gem *globalEndpointManager) GetReadEndpoints() ([]url.URL, error) {
+	return gem.locationCache.readEndpoints()
+}
+
+// GetAccountProperties retrieves account properties from the Cosmos DB instance.
+func (gem *globalEndpointManager) GetAccountProperties() (accountProperties, error) {
+	operationContext := pipelineRequestOptions{
+		resourceType:    resourceTypeDatabaseAccount,
+		resourceAddress: "",
+	}
+
+	path, err := generatePathForNameBased(resourceTypeDatabaseAccount, "", false)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to generate path for name-based request: %v", err)
+	}
+
+	azResponse, err := gem.client.sendGetRequest(path, context.Background(), operationContext, nil, nil)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to send GET request: %v", err)
+	}
+
+	properties, err := newAccountProperties(azResponse)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to parse account properties: %v", err)
+	}
+
+	return properties, nil
+}
+
+// newAccountProperties parses the account properties from the HTTP response.
+func newAccountProperties(azResponse *http.Response) (accountProperties, error) {
+	properties := accountProperties{}
+	err := azruntime.UnmarshalAsJSON(azResponse, &properties)
+	if err != nil {
+		return properties, err
+	}
+	return properties, nil
+}
+
+// GetLocation returns the location for the given endpoint from the location cache.
+func (gem *globalEndpointManager) GetLocation(endpoint url.URL) string {
+	return gem.locationCache.getLocation(endpoint)
+}
+
+// MarkEndpointUnavailableForRead marks an endpoint as unavailable for read operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForRead(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForRead(endpoint)
+}
+
+// MarkEndpointUnavailableForWrite marks an endpoint as unavailable for write operations.
+func (gem *globalEndpointManager) MarkEndpointUnavailableForWrite(endpoint url.URL) error {
+	return gem.locationCache.markEndpointUnavailableForWrite(endpoint)
+}
+
+// Update updates the location cache and the client's default endpoint based on the provided account properties.
+func (gem *globalEndpointManager) Update() error {
+	accountProperties, err := gem.GetAccountProperties()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve account properties: %v", err)
+	}
+
+	if err := gem.locationCache.update(accountProperties.WriteRegions, accountProperties.ReadRegions, gem.preferredRegions, &accountProperties.EnableMultipleWriteLocations); err != nil {
+		return fmt.Errorf("failed to update location cache: %v", err)
+	}
+	return nil
+}
+
+func (gem *globalEndpointManager) startBackgroundRefresh(refreshInterval time.Duration) {
```

Comment:
I think it would help to have a comment explaining the lifetime of the background refresh goroutine.  When does it start?  When should it stop?  If it stops, does it automatically get restarted?  etc...

Reply:
The goroutine lifetime is tied to the client lifetime, it should run until the client instance is destroyed/disposed.

Reply:
After our offline sync: The background refresh should be removed.